### DISCUSSION TITLE
Windows install Guide in here

### DISCUSSION THREAD

**Discussion by Rakly3** [imartinez/privateGPT#1195] [General]
# Install privateGPT Windows 10/11

### Clone the repo
- git clone https://github.com/imartinez/privateGPT
- cd privateGPT

### Create Conda env with Python 3.11
- conda create -n privateGPT python=3.11
- conda activate privateGPT

---
#### This can be entered as one command line
 - pip install python-dateutil>=2.8.2 tornado>=6.1 decorator pygments>=2.4.0 huggingface_hub pydantic uvicorn llama_index fastapi injector chromadb python-multipart gradio llama-cpp-python transformers torch

---
# Run in Windows Powershell terminal
- ### If Python installed from MS Store
  - (Invoke-WebRequest -Uri https://install.python-poetry.org -UseBasicParsing).Content | python -
- ### If Python install not from MS Store
  - (Invoke-WebRequest -Uri https://install.python-poetry.org -UseBasicParsing).Content | py -

---
# move file '_setup_' to parent folder
- copy **/privateGPT/scripts/setup** => main folder **/privateGPT/**

---
# some small tweaking
- go to **private_gpt/ui/** and open file **ui.py**
- In the code look for **upload_button = gr.components.UploadButton**
  - Change the value
    - **type="file"** => **type="filepath"**

- in the terminal enter **poetry run python -m private_gpt**
  - to use _other_ base than openAI _paid_ API chatGPT
  - in the main folder /privateGPT
  - manually change the values in _settings.yaml_ e.g. from _'mock'_ to _'local'_
    - _Or_ rename the files to make your choice _settings.yaml_ file-name.

**Comment by kalle07** [comment]
thy  ... but an error 

why they can not make an easy install like text-generation-webui   ;)

ok installl running all fine

error in powershell:
PS E:\privateGPT> C:\Users\kallemst\AppData\Roaming\Python\Scripts\poetry run python -m private_gpt
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "E:\privateGPT\private_gpt\__main__.py", line 3, in <module>
    import uvicorn
ModuleNotFoundError: No module named 'uvicorn'

error in ananconda setup (activated oc):
(privateGPT) E:\privateGPT>pip install python-dateutil>=2.8.2 tornado>=6.1 decorator pygments>=2.4.0 huggingface_hub pydantic uvicorn llama_index fastapi injector chromadb python-multipart gradio llama-cpp-python transformers torch

(privateGPT) E:\privateGPT>C:\Users\kallemst\AppData\Roaming\Python\Scripts\poetry run python -m private_gpt
21:15:45.661 [INFO    ] private_gpt.settings.settings_loader - Starting application with profiles=['default']
21:15:49.650 [INFO    ]         private_gpt.ui.ui - Mounting the gradio UI, at path=/
21:15:49.730 [INFO    ]             uvicorn.error - Started server process [16232]
21:15:49.730 [INFO    ]             uvicorn.error - Waiting for application startup.
21:15:49.730 [INFO    ]             uvicorn.error - Application startup complete.
21:15:49.732 [INFO    ]             uvicorn.error - Uvicorn running on http://0.0.0.0:8001 (Press CTRL+C to quit)


may 0.0.0.0 dont work on my machine, where i can change it to 127.0.0.1:8888
is some what else going wrong ?

OK SRY
if i manual 127.0.0.1:8001
the the browwser adress work but i cant do anything ...
no model (i hve copy one gguf inside "model" folder ...

must i install some api stuff
pip install gradio_client  ???



THX

**Comment by doobidoo** [comment]
The following section seems to be obsolete now:

> go to private_gpt/ui/ and open file ui.py
> 
> In the code look for upload_button = gr.components.UploadButton
> 
>     Change the value
>         type="file" => type="filepath"

**Reply by TomLucidor** [reply]
Why so? And what other tweaks should be anticipated?

**Comment by TomLucidor** [comment]
Notes on using `pyenv-win`:
```bat
:: [Info] ::  Mirror: https://www.python.org/ftp/python
pyenv-install: definition not found: 3.11

See all available versions with `pyenv install --list'.
```

**Reply by TomLucidor** [reply]
Remember to install the needed software through Chocolatey (so now it is 5-step process)
```
choco install git anaconda3
```
(Need to add PATH here)
```
git clone https://github.com/imartinez/privateGPT
cd privateGPT
conda create -n privateGPT python=3.11
conda activate privateGPT
pip install python-dateutil>=2.8.2 tornado>=6.1 decorator pygments>=2.4.0 huggingface_hub pydantic uvicorn llama_index fastapi injector chromadb python-multipart gradio llama-cpp-python transformers torch
```
(Install Poetry through pip, YAML file currently defaults to "local" btw so no need to sweat)

Side note: Oobabooga solved the `llama-cpp-python` issue with https://github.com/oobabooga/text-generation-webui/issues/1534#issuecomment-1579787872
See also: https://github.com/imartinez/privateGPT/discussions/336 https://github.com/imartinez/privateGPT/discussions/974
Addendum: WAIT is this is a versioning issue? https://github.com/imartinez/privateGPT/issues/683

Note 1: use Poetry that is installed through pip instead of `(Invoke-WebRequest -Uri https://install.python-poetry.org/ -UseBasicParsing).Content | py -` due to some kind of instability
Note 2: Only do `poetry run python -m private_gpt` after `poetry install --with ui` and `poetry run python scripts/setup` https://docs.privategpt.dev/installation

**Comment by kaas1551** [comment]
Hi.
I seem to run into an error regarding llama-cpp-python and CMake.
 im getting the following on Win 10:



error: subprocess-exited-with-error
× Building wheel for llama-cpp-python (pyproject.toml) did not run successfully.
│ exit code: 1
╰─> [20 lines of output]
    *** scikit-build-core 0.8.2 using CMake 3.28.3 (wheel)
    *** Configuring CMake...
    2024-03-18 05:18:51,019 - scikit_build_core - WARNING - Can't find a Python library, got libdir=None, ldlibrary=None, multiarch=None, masd=None
    loading initial cache file C:\Users\merij\AppData\Local\Temp\tmpo9sym148\build\CMakeInit.txt
    -- Building for: NMake Makefiles
    CMake Error at CMakeLists.txt:3 (project):
      Running
       'nmake' '-?'
      failed with:
       The system cannot find the file specified
    CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
    CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
    -- Configuring incomplete, errors occurred!
    *** CMake configuration failed
    [end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
ERROR: Failed building wheel for llama-cpp-python
ERROR: Could not build wheels for llama-cpp-python, which is required to install pyproject.toml-based projects

**Comment by kalle07** [comment]
again why such ascomplicate  (Chocolatey?)
i mean you can copy and paste from other that works much simpler and faster

spy on Jan he has made it ^^

**Reply by kaas1551** [reply]
What do you mean?

**Reply by kalle07** [reply]
sry, I meant that there is so much open source code here on Git, why don't you copy and paste or get an idea of how they did it ^^
like:
kobold
jan
lm-ai
oobadooga
...

**Comment by KarthikGowdaRamakrishna** [comment]
Same here i think 

Building wheels for collected packages: llama-cpp-python, ffmpy
  Building wheel for llama-cpp-python (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for llama-cpp-python (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      *** scikit-build-core 0.8.2 using CMake 3.28.3 (wheel)
      *** Configuring CMake...
      2024-03-19 15:16:05,179 - scikit_build_core - WARNING - Can't find a Python library, got libdir=None, ldlibrary=None, multiarch=None, masd=None
      loading initial cache file C:\Users\pavan\AppData\Local\Temp\tmp1hehjnor\build\CMakeInit.txt
      -- Building for: NMake Makefiles
      CMake Error at CMakeLists.txt:3 (project):
        Running

         'nmake' '-?'

        failed with:

         The system cannot find the file specified


      CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
      CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
      -- Configuring incomplete, errors occurred!

      *** CMake configuration failed
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for llama-cpp-python
  Building wheel for ffmpy (setup.py) ... done
  Created wheel for ffmpy: filename=ffmpy-0.3.2-py3-none-any.whl size=5605 sha256=62b33a049b9507c5027415a5819e2726da75f8176e170261750cbf62aaba905a   
  Stored in directory: c:\users\pavan\appdata\local\pip\cache\wheels\55\3c\f2\f6e34046bac0d57c13c7d08123b85872423b89c8f59bafda51
Successfully built ffmpy
Failed to build llama-cpp-python
ERROR: Could not build wheels for llama-cpp-python, which is required to install pyproject.toml-based projects

**Reply by kaas1551** [reply]
Make sure you install visual studio with the C++ workload aswell as CMake. This ended up fixing this particular issue for me. I still ended up running into other unrelated issues regardless of this, so I'm running my LLM's directly through Ollama (since i only needed the API and dont care for web ui)

**Reply by KarthikGowdaRamakrishna** [reply]
I'm pretty new to all this , sorry to ask this but how to install it? i tried to install CMake plugin on VS code extensions

**Reply by kaas1551** [reply]
VSCode is a code editor, entirely separate from Visual Studio, the IDE. You need to install visual studio. You also need MinGW. Please have a look here:

https://docs.privategpt.dev/installation/getting-started/installation#llama-cpp-known-issues-and-troubleshooting

**Comment by JTMarsh556** [comment]
This works for me:  Follow this video up until 6:07 [https://www.youtube.com/watch?v=hfnxVlPGLZs](url).  Then pick up here for ollama (I cant get good performance on this right now): [https://docs.privategpt.dev/installation/getting-started/installation#local-ollama-powered-setup-recommended](url) or here for for llama (works way better for me right now) [https://docs.privategpt.dev/installation/getting-started/installation#llama-cpp-support](url).  For CUDA support refer to the video first mentioned at 9:58 or [https://docs.privategpt.dev/installation/getting-started/installation#llama-cpp-windows-nvidia-gpu-support](url).  This will work if you follow all of the steps.
